### ISSUE TITLE
CI: Replace unmaintained cargo-script

### ISSUE THREAD

**Issue by jayvdb** [py2many/py2many#275]
It is unmaintained https://github.com/DanielKeep/cargo-script/issues/71

tl;dr: See if rust-script will add clippy mode, otherwise try cargo-play, or fall back to cargo-eval.

https://github.com/reitermarkus/cargo-eval is a slightly more active fork, with 59 extra commits, and an extra release https://github.com/reitermarkus/cargo-eval/releases/tag/v0.1.0 dated 22 Aug 2019 and 16 unreleased commits since.

https://github.com/fornwall/rust-script is actively developed, and we were using it until 7b4aef55, however we are using cargo-script's `--build-only --debug` flags to separate compile vs invocation errors, this is not a good fit because it doesnt have a similar concept.  I've created an issue in that project for a Clippy mode which would suffice.

https://github.com/stevedonovan/runner was considered when first building the test rig, however I ran into issues with cargo dependencies.  The bug has been fixed, but `cargo-deps`/inline `cargo` not yet supported (c.f. https://github.com/stevedonovan/runner/issues/7), and I think that is needed before this could be re-considered.  It has `--compile-only` which could be a suitable replacement for `--build-only`.  It also has a boilerplate wrapper around the invoked code which will mask some syntax errors, but has a `--no-prelude` opt to disable this.

https://github.com/fanzeyi/cargo-play has a similar mechanism for including dependencies in the script, but lacks full "inline" `cargo`.  It also allows multiple files to be merged together which could be a temporary shortcut towards https://github.com/adsharma/py2many/issues/158.  It has args `--check`, which may be like `--build-only`, and `--cargo-option` which might make it possible to run clippy, which would be good enough for a "compile-only" stage.  `--cargo-option` probably negates the need for "inline" `cargo`, as "edition = 2018" is constant, so it doesnt need to be inline in each script.

https://github.com/igor-petruk/scriptisto is another suggested replacement, and it says it supports any language.  Last release 16 Nov 2020.  It has templates for a few languages.
It doesnt have any special mechanism for Cargo dependencies.  We would need to build our own template with this, similar to https://github.com/igor-petruk/scriptisto/blob/master/data/templates/rust.rs, with all of the possible deps included, or generate a template for each bin-script with the subset of dependencies needed.  It does have a `build` command, which can be a replacement for `--build-only`.

**Comment by jayvdb** [comment]
It is now failing to build for me, on macos monterey
```
error: could not compile `libc`

Caused by:
  process didn't exit successfully: `rustc --crate-name libc /Users/jayvdb/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.122/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C linker-plugin-lto --cfg 'feature="default"' --cfg 'feature="std"' -C metadata=be51ef3f6d857fbf -C extra-filename=-be51ef3f6d857fbf --out-dir /var/folders/j6/q46zbprs12n5ds9z7zytq5mm0000gn/T/cargo-installdxLfF5/release/deps -L dependency=/var/folders/j6/q46zbprs12n5ds9z7zytq5mm0000gn/T/cargo-installdxLfF5/release/deps --cap-lints allow --cfg freebsd11 --cfg libc_priv_mod_use --cfg libc_union --cfg libc_const_size_of --cfg libc_align --cfg libc_int128 --cfg libc_core_cvoid --cfg libc_packedN --cfg libc_cfg_target_vendor --cfg libc_non_exhaustive --cfg libc_ptr_addr_of --cfg libc_underscore_const_names` (signal: 6, SIGABRT: process abort signal)
```

**Comment by jayvdb** [comment]
`cargo-eval` adopted in https://github.com/py2many/py2many/pull/489
